pyproject.toml

```diff
@@ -0,0 +1,40 @@
+[project]
+name = "torchmd-net"
+description = "TorchMD-Net package"
+authors = [{ name = "CompScienceLab", email = "gianni.defabritiis@upf.edu" }]
+readme = "README.md"
+license = { file = "LICENSE" }
+requires-python = ">=3.8"
+dynamic = ["version"]
+classifiers = [
+  "Programming Language :: Python :: 3",
+  "Operating System :: POSIX :: Linux",
+]
+dependencies = []
+
+[project.urls]
+"Homepage" = "https://github.com/torchmd/torchmd-net"
+"Bug Tracker" = "https://github.com/torchmd/torchmd-net/issues"
+
+[project.scripts]
+torchmd-train = "torchmdnet.scripts.train:main"
+
+[tool.setuptools.packages.find]
+where = [""]
+include = ["torchmdnet*"]
+namespaces = false
+
+[tool.setuptools.package-data]
+torchmdnet = ["extensions/torchmdnet_extensions.so"]
+
+[tool.versioneer]
+VCS = "git"
+style = "pep440"
+versionfile_source = "torchmdnet/_version.py"
+versionfile_build = "torchmdnet/_version.py"
+tag_prefix = ""
+parentdir_prefix = "torchmdnet-"
+
+[build-system]
+requires = ["setuptools", "toml", "versioneer[toml]==0.28", "torch<2.2"]
```

Comment:
There are probably more dependecies, is it required that they are listed twice? (environment.yaml and here) I feel like this will get desynchronized rather quickly.
Also I do not think there is a restriction on torch version.

Reply:
These are the requirements for _building_ the package, not for running it. So practically everything that's required to run `setup.py`

Reply:
The restriction of torch version I took from the environment.yaml but no issues on my side with removing it. I doubt it affects the package building

Reply:
Yeah I am actually seeing that issue right now where it's trying to find CUDA but it's not installed in the venv. I guess we will have to ditch the venv approach. But we can use versioneer regardless. Let me fix it

Reply:
I got PTSD from when I was trying to port NNPOps to pip and learned a bit about pyproject.toml. In my experience the venv thing will just not ever work. It assumes everything you need is in pip AND that packages are compatible between versions of the dependencies. I often ran into the build env installing torch X but the run env getting torch Y.
 I cannot fathom what they were thinking when they designed it that way. I can only assume I am missing something fundamental about it -.-

Reply:
You can still use pyproject.toml without venvs though I think. You just need to specify the correct build engine but I'm not going to bother. This works right now so let's stick to that :P